plugin/storage/grpc/shared/grpc_client.go

```diff
@@ -34,6 +35,18 @@ type grpcClient struct {
 	depsReaderClient storage_v1.DependenciesReaderPluginClient
 }
 
+// updateContextWithBearerToken updates the outgoing context with a bearer token extracted from the source
+func (c *grpcClient) updateContextWithBearerToken(outgoing context.Context, source context.Context) context.Context {
```

Comment:
I am not sure what would be a cleaner way to handle this. The problem here is that `FindTraces` and `FindTraceIDs` uses `context.Background()` so this function has to operate on two contexts.